node/src/db_config/persistent_configuration.rs

```diff
@@ -415,19 +415,25 @@ impl PersistentConfiguration for PersistentConfigurationReal {
         self.simple_set_method("start_block", value)
     }
 
-    fn max_block_count(&self) -> Result<u64, PersistentConfigError> {
+    fn max_block_count(&self) -> Result<Option<u64>, PersistentConfigError> {
         match self.get("max_block_count") {
             Ok(max_block_count) => match decode_u64(max_block_count) {
```

Comment:
Here you're supposed to take advantage of the return type of `decode_u64`. It actually returns a result of an `Option`.
In that case it's preferred that you'd treat it simply by returning None immediately and so you can solve this by deleting this whole line:

```
Err(TypedConfigLayerError::BadNumberFormat(value)) if value.is_empty() => Ok(None),
```
The first match close should take care for the rest.